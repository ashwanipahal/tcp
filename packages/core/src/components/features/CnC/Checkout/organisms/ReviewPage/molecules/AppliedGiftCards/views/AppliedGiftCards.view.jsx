import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import { getLabelValue } from '@tcp/core/src/utils';
import styles from '../styles/AppliedGiftCards.style';

const getGiftCard = giftCard => ({
  id: giftCard.get('id'),
  endingNumbers: giftCard.get('endingNumbers'),
  remainingBalance: giftCard.get('remainingBalance'),
});

const renderGiftCard = (labels, endingNumbers, remainingBalance) =>
  `${getLabelValue(
    labels,
    'lbl_review_appliedGiftCardEndingIn'
  )} ${endingNumbers} | ${getLabelValue(
    labels,
    'lbl_review_appliedGiftCardRemainingBal'
  )}: $${remainingBalance.toFixed(2)}`;

const renderGiftCardMessage = labels => (
  <BodyCopy component="p" className="headsup-container">
    <BodyCopy
      component="span"
      fontSize="fs16"
      color="gray[900]"
      fontFamily="secondary"
      fontWeight="extrabold"
    >
      {getLabelValue(labels, 'lbl_review_giftCardHeadsup')}
    </BodyCopy>
    <BodyCopy component="span" fontSize="fs16" color="gray[900]" fontFamily="secondary">
      {` ${getLabelValue(labels, 'lbl_review_giftCardMessage')}`}
    </BodyCopy>
  </BodyCopy>
);

const AppliedGiftCards = ({ className, appliedGiftCards, labels }) => {
  return (
    <BodyCopy component="div" className={`${className} applied-gift-cards`}>
      {appliedGiftCards && appliedGiftCards.size > 0
        ? appliedGiftCards.map(giftCard => {
            const { id, endingNumbers, remainingBalance } = getGiftCard(giftCard);
            return (
              <BodyCopy
                component="div"
                fontSize="fs16"
                fontFamily="secondary"
                color="gray[700]"
                className="applied-gift-card"
                key={id}
              >
                {renderGiftCard(labels, endingNumbers, remainingBalance)}
              </BodyCopy>
            );
          })
        : getLabelValue(labels, 'lbl_review_appliedGiftCardsNone')}
      {renderGiftCardMessage(labels)}
    </BodyCopy>
  );
};

AppliedGiftCards.propTypes = {
  appliedGiftCards: PropTypes.shape({}).isRequired,
  className: PropTypes.string.isRequired,
  labels: PropTypes.shape({
    lbl_review_appliedGiftCardEndingIn: PropTypes.string,
    lbl_review_appliedGiftCardRemainingBal: PropTypes.string,
    lbl_review_appliedGiftCardsNone: PropTypes.string,
    lbl_review_giftCardHeadsup: PropTypes.string,
    lbl_review_giftCardMessage: PropTypes.string,
  }),
};

AppliedGiftCards.defaultProps = {
  labels: {
    lbl_review_appliedGiftCardEndingIn: '',
    lbl_review_appliedGiftCardRemainingBal: '',
    lbl_review_appliedGiftCardsNone: '',
    lbl_review_giftCardHeadsup: '',
    lbl_review_giftCardMessage: '',
  },
};

export default withStyles(AppliedGiftCards, styles);
