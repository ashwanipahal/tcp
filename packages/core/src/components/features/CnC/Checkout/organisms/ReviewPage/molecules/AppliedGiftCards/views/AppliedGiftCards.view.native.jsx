import React from 'react';
import PropTypes from 'prop-types';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import { Container, GiftCardItem, GiftCardMessage } from '../styles/AppliedGiftCards.style.native';

/**
 * @function getGiftCard
 * @param {Map} giftCard
 * @description Converts gift card from immutable to normal object.
 */
const getGiftCard = giftCard => ({
  id: giftCard.get('id'),
  endingNumbers: giftCard.get('endingNumbers'),
  remainingBalance: giftCard.get('remainingBalance'),
});

/**
 * @function renderGiftCard
 * @param {Object} labels
 * @param {String} endingNumbers
 * @param {Number} remainingBalance
 * @description This method renders the applied gift card row.
 * @returns {String}
 */
const renderGiftCard = (labels, endingNumbers, remainingBalance) =>
  `${labels.lbl_review_appliedGiftCardEndingIn} ${endingNumbers} | ${
    labels.lbl_review_appliedGiftCardRemainingBal
  }: $${remainingBalance.toFixed(2)}`;

/**
 * @function renderGiftCardMessage
 * @param {Object} labels
 * @description This method renders Gift card heads up message
 * @returns {JSX}
 */
const renderGiftCardMessage = labels => (
  <GiftCardMessage>
    <BodyCopy
      fontSize="fs16"
      color="gray.900"
      mobileFontFamily="secondary"
      fontWeight="extrabold"
      text={labels.lbl_review_giftCardHeadsup}
    />
    <BodyCopy
      fontSize="fs16"
      color="gray.900"
      mobileFontFamily="secondary"
      text={` ${labels.lbl_review_giftCardMessage}`}
    />
  </GiftCardMessage>
);

/**
 * @function AppliedGiftCards
 * @param {Object} props
 * @description Applied Gift cards Functional component
 * @returns {JSX}
 */
export const AppliedGiftCards = ({ appliedGiftCards, labels }) => {
  const isGiftCardAdded = appliedGiftCards && appliedGiftCards.size > 0;
  return (
    <Container>
      {isGiftCardAdded ? (
        appliedGiftCards.map(giftCard => {
          const { id, endingNumbers, remainingBalance } = getGiftCard(giftCard);
          return (
            <GiftCardItem>
              <BodyCopy
                fontSize="fs16"
                mobileFontFamily="secondary"
                color="gray.700"
                key={id}
                text={renderGiftCard(labels, endingNumbers, remainingBalance)}
              />
            </GiftCardItem>
          );
        })
      ) : (
        <BodyCopy
          fontSize="fs16"
          mobileFontFamily="secondary"
          color="gray.700"
          text={labels.lbl_review_appliedGiftCardsNone}
        />
      )}
      {isGiftCardAdded && renderGiftCardMessage(labels)}
    </Container>
  );
};

AppliedGiftCards.propTypes = {
  appliedGiftCards: PropTypes.shape({}).isRequired,
  labels: PropTypes.shape({
    lbl_review_appliedGiftCardEndingIn: PropTypes.string,
    lbl_review_appliedGiftCardRemainingBal: PropTypes.string,
    lbl_review_appliedGiftCardsNone: PropTypes.string,
    lbl_review_giftCardHeadsup: PropTypes.string,
    lbl_review_giftCardMessage: PropTypes.string,
  }),
};

AppliedGiftCards.defaultProps = {
  labels: {
    lbl_review_appliedGiftCardEndingIn: '',
    lbl_review_appliedGiftCardRemainingBal: '',
    lbl_review_appliedGiftCardsNone: '',
    lbl_review_giftCardHeadsup: '',
    lbl_review_giftCardMessage: '',
  },
};

export default AppliedGiftCards;
