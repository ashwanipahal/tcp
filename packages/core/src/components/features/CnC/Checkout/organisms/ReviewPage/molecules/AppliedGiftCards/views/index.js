export { default } from './AppliedGiftCards.view';
