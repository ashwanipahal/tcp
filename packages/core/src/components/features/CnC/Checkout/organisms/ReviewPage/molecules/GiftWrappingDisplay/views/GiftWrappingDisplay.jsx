import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '../../../../../../../../common/hoc/withStyles';
import styles from '../styles/GiftWrappingDisplay.style';
import BodyCopy from '../../../../../../../../common/atoms/BodyCopy';

class GiftWrappingDisplay extends React.PureComponent {
  render() {
    const { displayName, className, labels } = this.props;
    return (
      <div className={className}>
        <div className="gift-wrapping-container">
          <BodyCopy
            fontSize="fs16"
            fontFamily="secondary"
            color="gray.900"
            fontWeight="extrabold"
            dataLocator="gift-wrapping-title"
          >
            {labels.lbl_review_sectionShippingGiftServiceTitle}
          </BodyCopy>
          <BodyCopy
            fontSize="fs16"
            fontFamily="secondary"
            color="gray.900"
            fontWeight="regular"
            dataLocator="gift-wrapping-name"
          >
            {displayName}
          </BodyCopy>
        </div>
      </div>
    );
  }
}

GiftWrappingDisplay.propTypes = {
  displayName: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  labels: PropTypes.shape({
    lbl_review_sectionShippingGiftServiceTitle: PropTypes.string,
  }),
};

GiftWrappingDisplay.defaultProps = {
  labels: {
    lbl_review_sectionShippingGiftServiceTitle: 'Gift Services',
  },
};

export default withStyles(GiftWrappingDisplay, styles);
