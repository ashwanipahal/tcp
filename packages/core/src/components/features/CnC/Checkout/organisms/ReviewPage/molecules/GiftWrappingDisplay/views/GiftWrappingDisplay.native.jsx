import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import BodyCopy from '../../../../../../../../common/atoms/BodyCopy';
import style from '../styles/GiftWrappingDisplay.style.native';

const { GiftWrappingTitle, GiftWrappingName } = style;

class GiftWrappingDisplay extends React.PureComponent {
  render() {
    const { displayName, labels } = this.props;
    const { lbl_review_sectionShippingGiftServiceTitle: GiftServiceTitle } = labels;
    return (
      <View>
        <GiftWrappingTitle>
          <BodyCopy
            fontSize="fs16"
            mobileFontFamily="secondary"
            color="gray.900"
            fontWeight="extrabold"
            dataLocator="gift-wrapping-title"
            text={GiftServiceTitle}
          />
        </GiftWrappingTitle>
        <GiftWrappingName>
          <BodyCopy
            fontSize="fs16"
            mobileFontFamily="secondary"
            color="gray.900"
            fontWeight="regular"
            dataLocator="gift-wrapping-name"
            text={displayName}
          />
        </GiftWrappingName>
      </View>
    );
  }
}

GiftWrappingDisplay.propTypes = {
  displayName: PropTypes.string.isRequired,
  labels: PropTypes.shape({
    lbl_review_sectionShippingGiftServiceTitle: PropTypes.string,
  }),
};

GiftWrappingDisplay.defaultProps = {
  labels: {
    lbl_review_sectionShippingGiftServiceTitle: 'Gift Services',
  },
};

export default GiftWrappingDisplay;
export { GiftWrappingDisplay as GiftWrappingDisplayanilla };
