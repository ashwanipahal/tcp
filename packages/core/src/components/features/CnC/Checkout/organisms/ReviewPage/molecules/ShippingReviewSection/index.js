export { default } from './views/ShippingReviewSection';
