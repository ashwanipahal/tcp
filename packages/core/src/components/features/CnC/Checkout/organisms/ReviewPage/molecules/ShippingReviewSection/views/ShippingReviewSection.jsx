import React from 'react';
import PropTypes from 'prop-types';
import { formatPhoneNumber } from '../../../../../../../../../utils/formValidation/phoneNumber';
import withStyles from '../../../../../../../../common/hoc/withStyles';
import styles from '../styles/ShippingReviewSection.style';
import Row from '../../../../../../../../common/atoms/Row';
import Col from '../../../../../../../../common/atoms/Col';
import TitlePlusEditButton from '../../TitlePlusEditButton';
import Address from '../../../../../../../../common/molecules/Address';
import BodyCopy from '../../../../../../../../common/atoms/BodyCopy';
import ShippingMethodDisplay from '../../ShippingMethodDisplay';
import GiftWrappingDisplay from '../../GiftWrappingDisplay';
import ShipmentMethods from '../../../../../../common/molecules/ShipmentMethods';

export class ShippingReviewSection extends React.PureComponent {
  render() {
    const {
      className,
      shippingAddress,
      shippingMethod,
      isGiftOptionsEnabled,
      giftWrappingDisplayName,
      labels,
      onEdit,
      isExpressCheckout,
      shipmentMethods,
      formName,
      formSection,
    } = this.props;
    const {
      lbl_review_shippingSectionTitle: title,
      lbl_review_sectionAnchor: edit,
      lbl_review_sectionShippingAddressTitle: addressTitle,
      lbl_review_sectionShippingMethodTitle: shippingMethodTitle,
    } = labels;
    return (
      <div className={className} dataLocator="review-shipping-section">
        <Row fullBleed>
          <Col colSize={{ small: 6, medium: 8, large: 12 }}>
            <TitlePlusEditButton
              title={title}
              editTitle={edit}
              onEdit={onEdit}
              dataLocator="pickup-section"
            />
          </Col>
        </Row>

        <Row fullBleed>
          <Col colSize={{ small: 6, medium: 4, large: 5 }}>
            <div className="shippingAddressTitle">
              <BodyCopy
                fontSize="fs16"
                dataLocator=""
                fontFamily="secondary"
                color="gray.900"
                fontWeight="extrabold"
              >
                {addressTitle}
              </BodyCopy>
            </div>
            <Address className="addressStyle" address={shippingAddress.address} />
            <BodyCopy
              fontSize="fs16"
              dataLocator=""
              fontFamily="secondary"
              color="gray.900"
              fontWeight="regular"
            >
              {shippingAddress.emailAddress}
            </BodyCopy>
            {shippingAddress.phoneNumber && (
              <BodyCopy
                fontSize="fs16"
                dataLocator=""
                fontFamily="secondary"
                color="gray.900"
                fontWeight="regular"
              >
                {formatPhoneNumber(shippingAddress.phoneNumber)}
              </BodyCopy>
            )}
          </Col>
          <Col colSize={{ small: 6, medium: 4, large: 5 }}>
            {!isExpressCheckout && shippingMethod && (
              <ShippingMethodDisplay labels={labels} displayName={shippingMethod.displayName} />
            )}
            {isExpressCheckout && shippingMethod && (
              <ShipmentMethods
                shipmentMethods={shipmentMethods}
                formName={formName}
                formSection={formSection}
                selectedShipmentId={shippingMethod.id}
                shipmentHeader={shippingMethodTitle}
              />
            )}
            {isGiftOptionsEnabled && !isExpressCheckout && (
              <GiftWrappingDisplay labels={labels} displayName={giftWrappingDisplayName} />
            )}
          </Col>
        </Row>
        <Row fullBleed>
          <Col colSize={{ small: 6, medium: 4, large: 5 }}>
            {isExpressCheckout && (
              <GiftWrappingDisplay labels={labels} displayName={giftWrappingDisplayName} />
            )}
          </Col>
        </Row>
      </div>
    );
  }
}

ShippingReviewSection.propTypes = {
  className: PropTypes.string.isRequired,
  labels: PropTypes.shape({}),
  isGiftOptionsEnabled: PropTypes.bool,
  giftWrappingDisplayName: PropTypes.string,
  shippingAddress: PropTypes.shape({}),
  shippingMethod: PropTypes.shape({
    id: PropTypes.string.isRequired,
    displayName: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    shippingSpeed: PropTypes.string.isRequired,
    isDefault: PropTypes.bool,
  }).isRequired,
  onEdit: PropTypes.func.isRequired,
  isExpressCheckout: PropTypes.bool.isRequired,
  shipmentMethods: PropTypes.shape({}).isRequired,
  formName: PropTypes.string.isRequired,
  formSection: PropTypes.string.isRequired,
  expressReviewShippingSectionId: PropTypes.shape({}),
};

ShippingReviewSection.defaultProps = {
  labels: {},
  shippingAddress: {},
  isGiftOptionsEnabled: false,
  giftWrappingDisplayName: 'N/A',
  expressReviewShippingSectionId: {},
};

export default withStyles(ShippingReviewSection, styles);
export { ShippingReviewSection as ShippingReviewSectionvanilla };
