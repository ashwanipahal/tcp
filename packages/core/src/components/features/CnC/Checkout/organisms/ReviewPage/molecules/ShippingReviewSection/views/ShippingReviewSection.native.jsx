import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { formatPhoneNumber } from '@tcp/core/src/utils/formValidation/phoneNumber';
import GenericSkeleton from '@tcp/core/src/components/common/molecules/GenericSkeleton/GenericSkeleton.view.native';
import Address from '../../../../../../../../common/molecules/Address';
import BodyCopy from '../../../../../../../../common/atoms/BodyCopy';
import ShippingMethodDisplay from '../../ShippingMethodDisplay';
import GiftWrappingDisplay from '../../GiftWrappingDisplay';
import TitlePlusEditButton from '../../TitlePlusEditButton';
import style from '../styles/ShippingReviewSection.style.native';
import ShipmentMethods from '../../../../../../common/molecules/ShipmentMethods';

const {
  ShippingReviewContainer,
  AddressSection,
  AddressTitle,
  TitlePlusEditSection,
  SkeletonWrapper,
} = style;

export class ShippingReviewSection extends React.PureComponent {
  render() {
    const {
      shippingAddress,
      shippingMethod,
      isGiftOptionsEnabled,
      giftWrappingDisplayName,
      labels,
      onEdit,
      isExpressCheckout,
      shipmentMethods,
      formName,
      formSection,
      dispatch,
      expressReviewShippingSectionId,
      bagLoading,
    } = this.props;
    const {
      lbl_review_shippingSectionTitle: title,
      lbl_review_sectionAnchor: edit,
      lbl_review_sectionShippingAddressTitle: addressTitle,
      lbl_review_sectionShippingMethodTitle: shippingMethodTitle,
    } = labels;
    return (
      <>
        {!bagLoading ? (
          <>
            <TitlePlusEditSection>
              <TitlePlusEditButton
                title={title}
                editTitle={edit}
                onEdit={onEdit}
                dataLocator="shipping-section"
              />
            </TitlePlusEditSection>
            <ShippingReviewContainer>
              <View>
                <AddressTitle>
                  <BodyCopy
                    fontSize="fs16"
                    dataLocator=""
                    fontFamily="secondary"
                    color="gray.900"
                    fontWeight="extrabold"
                    text={addressTitle}
                  />
                </AddressTitle>
                <AddressSection>
                  {!!shippingAddress.address && (
                    <Address address={shippingAddress.address} regularName />
                  )}
                  <BodyCopy
                    fontSize="fs16"
                    dataLocator=""
                    fontFamily="secondary"
                    color="gray.900"
                    fontWeight="regular"
                    text={shippingAddress.emailAddress}
                  />
                  {!!shippingAddress.phoneNumber && (
                    <BodyCopy
                      fontSize="fs16"
                      dataLocator=""
                      fontFamily="secondary"
                      color="gray.900"
                      fontWeight="regular"
                      text={formatPhoneNumber(shippingAddress.phoneNumber)}
                    />
                  )}
                </AddressSection>
              </View>
              <View>
                {!isExpressCheckout && shippingMethod && (
                  <ShippingMethodDisplay labels={labels} displayName={shippingMethod.displayName} />
                )}
                {isExpressCheckout && shippingMethod && (
                  <ShipmentMethods
                    shipmentMethods={shipmentMethods}
                    formName={formName}
                    formSection={formSection}
                    selectedShipmentId={
                      expressReviewShippingSectionId &&
                      expressReviewShippingSectionId.shippingMethodId
                    }
                    shipmentHeader={shippingMethodTitle}
                    dispatch={dispatch}
                  />
                )}
                {isGiftOptionsEnabled && (
                  <GiftWrappingDisplay
                    labels={labels}
                    displayName={giftWrappingDisplayName}
                    onEdit={onEdit}
                    editTitle={edit}
                    isExpressCheckout={isExpressCheckout}
                  />
                )}
              </View>
            </ShippingReviewContainer>
          </>
        ) : (
          <>
            <TitlePlusEditSection>
              <TitlePlusEditButton title={title} dataLocator="shipping-section" />
            </TitlePlusEditSection>
            <SkeletonWrapper>
              <GenericSkeleton />
            </SkeletonWrapper>
          </>
        )}
      </>
    );
  }
}

ShippingReviewSection.propTypes = {
  labels: PropTypes.shape({}),
  isGiftOptionsEnabled: PropTypes.bool,
  giftWrappingDisplayName: PropTypes.string,
  shippingAddress: PropTypes.shape({}),
  shippingMethod: PropTypes.shape({
    id: PropTypes.string.isRequired,
    displayName: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    shippingSpeed: PropTypes.string.isRequired,
    isDefault: PropTypes.bool,
  }).isRequired,
  onEdit: PropTypes.func.isRequired,
  isExpressCheckout: PropTypes.bool.isRequired,
  bagLoading: PropTypes.bool.isRequired,
  shipmentMethods: PropTypes.shape({}).isRequired,
  formName: PropTypes.string.isRequired,
  formSection: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  expressReviewShippingSectionId: PropTypes.func.isRequired,
};

ShippingReviewSection.defaultProps = {
  labels: {},
  shippingAddress: {},
  isGiftOptionsEnabled: false,
  giftWrappingDisplayName: 'N/A',
};

export default ShippingReviewSection;
export { ShippingReviewSection as ShippingReviewSectionvanilla };
