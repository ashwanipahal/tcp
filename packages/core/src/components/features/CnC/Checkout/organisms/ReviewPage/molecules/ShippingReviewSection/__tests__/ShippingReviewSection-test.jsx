import React from 'react';
import { shallow } from 'enzyme';
import { ShippingReviewSectionvanilla } from '../views/ShippingReviewSection';

describe('GiftWrappingDisplay component', () => {
  it('should renders correctly props not present', () => {
    const props = { labels: {}, displayName: '' };
    const component = shallow(<ShippingReviewSectionvanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should renders correctly props are present', () => {
    const props = {
      className: '',
      labels: {
        lbl_review_sectionShippingMethodTitle: 'title',
      },
      displayName: 'Free',
      isGiftOptionsEnabled: true,
      shippingMethod: { displayName: '' },
    };
    const component = shallow(<ShippingReviewSectionvanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should renders correctly props are present and isexpresscheckout', () => {
    const props = {
      className: '',
      labels: {
        lbl_review_sectionShippingMethodTitle: 'title',
      },
      displayName: 'Free',
      isExpressCheckout: true,
      shippingMethod: { displayName: '' },
      shippingAddress: { phoneNumber: '1234567897' },
    };
    const component = shallow(<ShippingReviewSectionvanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
