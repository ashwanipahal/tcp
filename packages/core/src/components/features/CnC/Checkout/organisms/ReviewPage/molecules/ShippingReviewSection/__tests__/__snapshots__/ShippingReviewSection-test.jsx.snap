// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftWrappingDisplay component should renders correctly props are present 1`] = `
<div
  className=""
  dataLocator="review-shipping-section"
>
  <Styled(Row)
    fullBleed={true}
  >
    <Styled(Col)
      colSize={
        Object {
          "large": 12,
          "medium": 8,
          "small": 6,
        }
      }
    >
      <Styled(TitlePlusEditButton)
        bagLoading={false}
        dataLocator="pickup-section"
      />
    </Styled(Col)>
  </Styled(Row)>
  <Styled(AddressSkelton) />
</div>
`;

exports[`GiftWrappingDisplay component should renders correctly props are present and isexpresscheckout 1`] = `
<div
  className=""
  dataLocator="review-shipping-section"
>
  <Styled(Row)
    fullBleed={true}
  >
    <Styled(Col)
      colSize={
        Object {
          "large": 12,
          "medium": 8,
          "small": 6,
        }
      }
    >
      <Styled(TitlePlusEditButton)
        bagLoading={false}
        dataLocator="pickup-section"
      />
    </Styled(Col)>
  </Styled(Row)>
  <Styled(AddressSkelton) />
</div>
`;

exports[`GiftWrappingDisplay component should renders correctly props not present 1`] = `
<div
  dataLocator="review-shipping-section"
>
  <Styled(Row)
    fullBleed={true}
  >
    <Styled(Col)
      colSize={
        Object {
          "large": 12,
          "medium": 8,
          "small": 6,
        }
      }
    >
      <Styled(TitlePlusEditButton)
        bagLoading={false}
        dataLocator="pickup-section"
      />
    </Styled(Col)>
  </Styled(Row)>
  <Styled(AddressSkelton) />
</div>
`;
